Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -391,11 +400,43 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype._computeLightVertices = function (
       obstaclePolygons.push(obstacleHitBoxes[i][j]);
   }
 
+  var maxX = this._object.x + this._radius;
+  var minX = this._object.x - this._radius;
+  var maxY = this._object.y + this._radius;
+  var minY = this._object.y - this._radius;
+
   var flattenVertices = [];
-  for (var i = 0; i < obstaclePolygons.length; i++) {
+  for (var i = 1; i < obstaclePolygons.length; i++) {
     var vertices = obstaclePolygons[i].vertices;
     var verticesCount = vertices.length;
-    for (var j = 0; j < verticesCount; j++) flattenVertices.push(vertices[j]);
+    for (var j = 0; j < verticesCount; j++) {
+      flattenVertices.push(vertices[j]);
+
+      if (vertices[j][0] < minX) minX = vertices[j][0];
+      if (vertices[j][0] > maxX) maxX = vertices[j][0];
+      if (vertices[j][1] < minY) minY = vertices[j][1];
+      if (vertices[j][1] > maxY) maxY = vertices[j][1];
+    }
+  }
+
+  obstaclePolygons[0].vertices[0][0] = minX;
+  obstaclePolygons[0].vertices[0][1] = minY;
+  obstaclePolygons[0].vertices[1][0] = maxX;
+  obstaclePolygons[0].vertices[1][1] = minY;
+  obstaclePolygons[0].vertices[2][0] = maxX;
+  obstaclePolygons[0].vertices[2][1] = maxY;
+  obstaclePolygons[0].vertices[3][0] = minX;
+  obstaclePolygons[0].vertices[3][1] = maxY;
+
+  var boundingSquareHalfDiag = Math.max(
+    Math.hypot(this._object.x - minX, this._object.y - minY),
+    Math.hypot(maxX - this._object.x, this._object.y - minY),
+    Math.hypot(maxX - this._object.x, maxY - this._object.y),
+    Math.hypot(this._object.x - minX, maxY - this._object.y)
```

Comment:
Slight optimisation to avoid computing 4 times the square root instead of 1. Don't use Math.hypot, instead compute the values before taking the square root with Math.pow. Example, do for the first line just:
Math.pow(this._object.x - minX, 2) + Math.pow(this._object.y - minY, 2)

(this is what Math.hypot is doing internally), then take the max (Math.max) and finally Math.sqrt the result.

This works because `0 < x < y <=> sqrt(x) < sqrt(y)`